tests/testthat/test-fit_model.R

```diff
@@ -0,0 +1,189 @@
+    seed = 12345,
+    horizon = 0,
+    priors = priors,
+    sampler = sampler_opts
+  )
+
+  expect_s3_class(fit, "epinow")
+})
+
+test_that("Minimal model fit with no right trunc or delay runs", {
+  # Parameters
+  parameters <- list(
+    generation_interval = gostic_gt_pmf,
+    delay_interval = NA,
+    right_truncation = NA
+  )
+  # Data -- 5 points only
+  data_path <- test_path("data", "test_data.parquet")
+  con <- DBI::dbConnect(duckdb::duckdb())
+  data <- DBI::dbGetQuery(con, "
+                         SELECT
+                           report_date,
+                           reference_date,
+                           disease,
+                           geo_value AS state_abb,
+                           value AS confirm
+                         FROM read_parquet(?)
+                         ORDER BY reference_date
+                         LIMIT 5
+                          ",
+    params = list(data_path)
+  )
+  DBI::dbDisconnect(con)
+  # Priors
+  priors <- list(
+    rt = list(
+      mean = 1,
+      sd = 0.2
+    ),
+    gp = list(
+      alpha_sd = 0.05
+    )
+  )
+  # Sampler
+  sampler_opts <- list(
+    cores = 1,
+    chains = 1,
+    adapt_delta = 0.8,
+    max_treedepth = 10,
+    iter_warmup = 100,
+    iter_samples = 100
+  )
+
+  fit <- fit_model(
+    data = data,
+    parameters = parameters,
+    seed = 12345,
+    horizon = 0,
+    priors = priors,
+    sampler = sampler_opts
+  )
+
+  expect_s3_class(fit, "epinow")
+})
+
+test_that("Bad params w/ failing fit issues warning and returns NA", {
+  # Parameters
+  parameters <- list(
+    generation_interval = gostic_gt_pmf,
+    delay_interval = NA,
+    right_truncation = NA
+  )
+  # Data -- 5 points only
+  data_path <- test_path("data", "test_data.parquet")
+  con <- DBI::dbConnect(duckdb::duckdb())
+  data <- DBI::dbGetQuery(con, "
+                         SELECT
+                           report_date,
+                           reference_date,
+                           disease,
+                           geo_value AS state_abb,
+                           value AS confirm
+                         FROM read_parquet(?)
+                         ORDER BY reference_date
+                         LIMIT 5
+                          ",
+    params = list(data_path)
+  )
+  DBI::dbDisconnect(con)
+  # Priors
+  priors <- list(
+    rt = list(
+      mean = 1,
+      sd = 0.2
+    ),
+    gp = list(
+      alpha_sd = 0.05
+    )
+  )
+  # Sampler
+  sampler_opts <- list(
+    cores = 1,
+    chains = 1,
+    adapt_delta = 0.8,
+    max_treedepth = 10,
+    iter_warmup = -100,
+    iter_samples = 100
+  )
+
+  expect_warning(
+    fit <- fit_model(
+      data = data,
+      parameters = parameters,
+      seed = 12345,
+      horizon = 0,
+      priors = priors,
+      sampler = sampler_opts
+    ),
+    class = "failing_fit"
+  )
+
+  expect_true(is.na(fit))
+})
+
+test_that("Right truncation longer than data throws error", {
```

Comment:
Do we want a test for this? To me this is more of a problem in EpiNow2, that was [fixed](https://github.com/epiforecasts/EpiNow2/issues/735). Seems like it might be something we check for and warn about in format_right_truncation, but idk if we need a test for this?

Reply:
I think we need the check in the pipeline because we haven't bumped the EpiNow2 version yet, so the bug is still present. And because the functionality is there, we need the test for it.